include/PriorityQueue.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <iostream>
+#include <list>
+#include <exception>
+#include "Dog.h"
+
+struct MyComparator {
+    template<typename T>
+    bool operator()(const T& a, const T& b) const {

```

Comment:
There no need to separate the method declaration and the method definition. You could just put them together.